File: 1/1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
        
    let res = [];
    
    for(let i = 0; i < nums.length; i++){     
        let diff = target - nums[i];
        
        if(diff in res){
            return [res[diff], i];
        }        
        res[nums[i]] = i;        
    }
      
};

File: 0001/1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> numList = new HashMap<>();
        for(int i = 0;i< nums.length;i++){
            if(numList.containsKey(target-nums[i]))
                return new int[] {i, numList.get(target - nums[i])};
            numList.put(nums[i],i);
            
        }
        return new int[2];
    }
}